### PR TITLE
Add type information to the Metrics MBeans

### LINKED ISSUE #16729: Beans missing Java type
**Describe the bug**

It appears that all beans of type `Metrics` are missing the [standard](https://www.oracle.com/technetwork/java/javase/tech/best-practices-jsp-136021.html) type information that many monitoring solutions rely on.

Capture

### PR DESCRIPTION
**Pull request by blazember**
Fixes #16729

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/jmx/JmxPublisherTest.java [hunk 1]
```diff
@@ -285,4 +290,17 @@ public void when_shutdown_noMBeansLeak() throws Exception {
     private MetricDescriptor newDescriptor() {
         return DEFAULT_DESCRIPTOR_SUPPLIER.get();
     }
+
+    private BiTuple<String, List<TriTuple<String, Number, Type>>> metric(String objectName,
+                                                                           List<TriTuple<String, Number, Type>> measurements) {
```
Comment (puzpuzpuz) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/jmx/JmxPublisherTest.java]: nit: padding seems wrong here (2 extra spaces)

**Comment by puzpuzpuz** [comment]
Test failure seems to be related with #16206

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/jmx/JmxPublisherTest.java (hunk 1 above)
Comment (blazember) [reply hazelcast/src/test/java/com/hazelcast/internal/metrics/jmx/JmxPublisherTest.java]: Hm :thinking: Lazy IDE :smile: Thanks, fixed here and in the other PR too.

**Comment by blazember** [comment]
Thanks for the review @mmedenjak @puzpuzpuz :+1:

**Pull request merged**